server-spi-private/src/main/java/org/keycloak/userprofile/UserProfileProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.userprofile;
+
+import org.keycloak.userprofile.util.UpdateUserProfileContext;
+import org.keycloak.userprofile.validation.UserProfileValidationResult;
+import org.keycloak.provider.Provider;
+import org.keycloak.provider.ProviderFactory;
+import org.keycloak.representations.idm.UserAttributesRepresentation;
+import org.keycloak.userprofile.validation.ValidationScenario;
+
+/**
+ * @author <a href="mailto:markus.till@bosch.io">Markus Till</a>
+ */
+public interface UserProfileProvider extends Provider, ProviderFactory<UserProfileProvider> {
+
+    UserProfileValidationResult validate(UserAttributesRepresentation rep, UpdateUserProfileContext user, ValidationScenario validationScenario);
```

Comment:
I will do a refactoring of this solution.
But I would prefer to introduce no"empty" value UserProfiles for getCurrent() as it would be quite complex to give the profile the ability to estimate how to create a user in the given context. Therefore I would expect that the caller always initialized the new user before calling the validation function.

Reply:
@stianst please have a look if the current interface is more as expected.